index.bs

```diff
@@ -65,16 +65,19 @@ beyond those described in the Generic Sensor API [[!GENERIC-SENSOR]].
 Model {#model}
 =====
 
-The orientation sensors's associated {{Sensor}} subclass is the {{DeviceOrientationSensor}} class.
+The device orientation sensors's associated {{Sensor}} subclass is the {{DeviceOrientationSensor}} class.
 
-Orientation sensor is a [=high-level=] sensor which is created through [=sensor-fusion=]
-of [=low-level=] motion sensors (accelerometer, gyroscope, magnetometer).
+The device orientation sensor is a [=high-level=] sensor which is created through [=sensor-fusion=]
+of the [=low-level=] motion sensors (accelerometer, gyroscope). The type of fusion (complementary
```

Comment:
"Game" orientation vector and "Geographic" orientation vectors (for lack of better terminology for now), are two distinct fused sensors, based on different sensors (gyroscope for the former, generally accelerometer and magnetometer for the latter), with different characteristics (low latency, drift for the former, higher latency, north-bound, no drift, and lower battery consumption for the latter), different use case (VR, gaming, and gestures for the former), (AR and indoor location for the latter), and a similar data format (but for different data).

The current DeviceOrientationEvent spec conflates both, which has tripped countless generations of developers and implementors, and which is one of the chore reasons for the complete redesign of the sensor specs. It would be really regrettable to repeat the same mistake here. **These should really need be two different sensor types** (i.e. two different constructors).

Platforms which cannot provide both sensor types are clearly missing out on a bunch of use cases, and developers need to be aware of this (that is, you really don't want to silently replace one sensor type by another on such platforms). It might be worth investigating how to provide custom fusion on such platforms, in C++.